src/bp/lang-server/service/index.ts

```diff
@@ -3,22 +3,46 @@ import fs from 'fs'
 import _ from 'lodash'
 import lru from 'lru-cache'
 import ms from 'ms'
+import os from 'os'
 import path from 'path'
+import process from 'process'
 import { VError } from 'verror'
-
 import toolkit from '../../ml/toolkit'
 import Logger from '../../simple-logger'
-
+import bytes from 'bytes'
 import { LoadedBPEModel, LoadedFastTextModel, ModelFileInfo, ModelSet } from './typing'
 
+interface RamInfos {
+  free: number
+  total: number
+  prediction: number
+}
+
 const maxAgeCacheInMS = ms('24h')
 
+const MODEL_SAFETY_BUFFER = 0.2
+const RAM_SAFETY_THRESHOLD = 0.01
+
 // Examples:  "scope.en.300.bin" "bp.fr.150.bin"
 const FAST_TEXT_MODEL_REGEX = /^(\w+)\.(\w+)\.(\d+)\.bin$/i
 
 // Examples: "scope.en.bpe.model" "bp.en.bpe.model"
 const BPE_MODEL_REGEX = /^(\w+)\.(\w+)\.bpe\.model$/i
 
+const getModelSize = (dims: number) => {
```

Comment:
This is not a model size getter. It's a model size **estimator**. Something like `estimateModelSize()` is more accurate.

\+ why is this not a private method?

\+ what I meant was to extract alpha and beta in constants with capital case at the top of the file (This is a a widespread convention in cs)

\+ Usually, we don't call a function just alpha or beta... We try to name variables by what they actually represent:

```
alpha => MODEL_MB_PER_DIM ?
beta => MODEL_CONSTANT_MB ?
``` 

How did you estimated this ? Typically, if you've took some data points and estimated a linear regression model, I expect the PR description to include your scientific process (with a graph).